### ISSUE TITLE
Lehrerkürzel sind nicht immer Kürzel, werden aber so dargestellt

### ISSUE THREAD

**Issue by lorax42** [vplan-fr/VPlan_FR#305]
In der Infobox wird das Wort 'Die' jedesmal als Lehrerkürzel verlinkt, da ein Lehrer halt dieses Kürzel hat.
![Screenshot from 2024-08-04 15-58-46](https://github.com/user-attachments/assets/abdcd8c0-3c2e-4090-a738-9adf651b9a49)

**Comment by lorax42** [comment]
Vielleicht könnte man die Sätze analysieren und das 'Die' nicht markieren, wenn es ein Artikel ist. Beziehungsweise auch einfach am Satzanfang, wobei man sehen müsste, wie viele false positives das erzeugt.

**Comment by lorax42** [comment]
```c
for (size_t i = 0; i < info.size(); i++){
    if (info[i] == '.'){
        if (info[i+1] == ' '){
            // then don't highlight following 'Die' ...
        }
    }
}
```

**Comment by ArtrenH** [comment]
Aber es kann ja auch sowas wie
"Heute irgendein Event. Die ist verantwortlich" in der Info stehen und das meint dann halt das Kürzel und das irgendwas mit dem Lehrer los ist.
Also eigentlich müsste man da jedes Mal ein AI Language Model fragen, ob das ein Kürzel oder ein Wort ist.

**Comment by lorax42** [comment]
Ja, hatte ich auch schon überlegt, aber das wäre ziemlich umständig. Ich glaube, dass das vielleicht selten genug vorkommt, dass man ein Auge zudrücken kann. Im Durchschnitt würde es halt wahrscheinlich trotzdem akkurater sein.

**Comment by lorax42** [comment]
Man bräuchte nicht mal ein großes AI Modell. Also kein LLM oder so. Vielleicht wäre es klein genug lokal laufen zu lassen, damit durch den API-Key, den man z.B. für Chat-GPT brauch, keine Kosten entstehen. Sowas gibt's bestimmt kostenlos. Als FOSS oder zumindest open weights (wie LLAMA nur für Satzanalyse).
